proximo-server/nats-streaming.go

```diff
@@ -14,17 +14,29 @@ import (
 )
 
 type natsStreamingHandler struct {
-	url       string
 	clusterID string
+	nc        *nats.Conn
+}
+
+func NewNatsStreamingHandler(url, clusterID string) (*natsStreamingHandler, error) {
+	nc, err := nats.Connect(url, nats.Name("proximo-nats-streaming-"+generateID()))
```

Comment:
what is the nats.Name

Reply:
that's what the option is called, it modifies the options object, it's the functional ...optionFunc pattern